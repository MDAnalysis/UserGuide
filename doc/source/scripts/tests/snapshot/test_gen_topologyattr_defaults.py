from unittest.mock import patch

from gen_topologyattr_defaults import TopologyDefaults


def test_TopologyDefaults(snapshot):
    with patch("builtins.open"):
        td = TopologyDefaults()
    assert td.table_writer.lines == snapshot
