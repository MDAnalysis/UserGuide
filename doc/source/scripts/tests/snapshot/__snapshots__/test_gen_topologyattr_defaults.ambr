# serializer version: 1
# name: test_TopologyDefaults
  list([
    list([
      'altLocs',
      'altLoc',
      "''",
      'atom',
      <class 'object'>,
    ]),
    list([
      'angles',
      'angles',
      'No default values',
      'atom',
      None,
    ]),
    list([
      'aromaticities',
      'aromaticity',
      'False',
      'atom',
      <class 'bool'>,
    ]),
    list([
      'atomiccharges',
      'atomiccharge',
      'No default values',
      'atom',
      None,
    ]),
    list([
      'atomnums',
      'atomnum',
      'No default values',
      'atom',
      None,
    ]),
    list([
      'bonds',
      'bonds',
      'No default values',
      'atom',
      None,
    ]),
    list([
      'chainIDs',
      'chainID',
      "''",
      'atom',
      <class 'object'>,
    ]),
    list([
      'chargegroups',
      'chargegroup',
      'No default values',
      'atom',
      None,
    ]),
    list([
      'charges',
      'charge',
      '0.0',
      'atom',
      <class 'float'>,
    ]),
    list([
      'chiralities',
      'chirality',
      'No default values',
      'atom',
      'U1',
    ]),
    list([
      'cmaps',
      'cmaps',
      'No default values',
      'atom',
      None,
    ]),
    list([
      'dihedrals',
      'dihedrals',
      'No default values',
      'atom',
      None,
    ]),
    list([
      'elements',
      'element',
      "''",
      'atom',
      <class 'object'>,
    ]),
    list([
      'epsilon14s',
      'epsilon14',
      '0.0',
      'atom',
      <class 'float'>,
    ]),
    list([
      'epsilons',
      'epsilon',
      '0.0',
      'atom',
      <class 'float'>,
    ]),
    list([
      'formalcharges',
      'formalcharge',
      '0.0',
      'atom',
      <class 'int'>,
    ]),
    list([
      'gbscreens',
      'gbscreen',
      '0.0',
      'atom',
      <class 'float'>,
    ]),
    list([
      'icodes',
      'icode',
      "''",
      'residue',
      <class 'object'>,
    ]),
    list([
      'ids',
      'id',
      'continuous sequence from 1 to n_atoms',
      'atom',
      <class 'int'>,
    ]),
    list([
      'impropers',
      'impropers',
      'No default values',
      'atom',
      None,
    ]),
    list([
      'masses',
      'mass',
      '0.0',
      'atom',
      <class 'numpy.float64'>,
    ]),
    list([
      'models',
      'model',
      'No default values',
      'segment',
      None,
    ]),
    list([
      'molnums',
      'molnum',
      'No default values',
      'residue',
      <class 'numpy.int64'>,
    ]),
    list([
      'moltypes',
      'moltype',
      "''",
      'residue',
      <class 'object'>,
    ]),
    list([
      'names',
      'name',
      "''",
      'atom',
      <class 'object'>,
    ]),
    list([
      'nbindices',
      'nbindex',
      '0',
      'atom',
      <class 'int'>,
    ]),
    list([
      'occupancies',
      'occupancy',
      '0.0',
      'atom',
      <class 'float'>,
    ]),
    list([
      'radii',
      'radius',
      '0.0',
      'atom',
      <class 'float'>,
    ]),
    list([
      'record_types',
      'record_type',
      "'ATOM'",
      'atom',
      <class 'object'>,
    ]),
    list([
      'resids',
      'resid',
      'continuous sequence from 1 to n_residues',
      'residue',
      <class 'int'>,
    ]),
    list([
      'resnames',
      'resname',
      "''",
      'residue',
      <class 'object'>,
    ]),
    list([
      'resnums',
      'resnum',
      'continuous sequence from 1 to n_residues',
      'residue',
      <class 'int'>,
    ]),
    list([
      'rmin14s',
      'rmin14',
      '0.0',
      'atom',
      <class 'float'>,
    ]),
    list([
      'rmins',
      'rmin',
      '0.0',
      'atom',
      <class 'float'>,
    ]),
    list([
      'segids',
      'segid',
      "''",
      'segment',
      <class 'object'>,
    ]),
    list([
      'solventradii',
      'solventradius',
      '0.0',
      'atom',
      <class 'float'>,
    ]),
    list([
      'tempfactors',
      'tempfactor',
      '0.0',
      'atom',
      <class 'float'>,
    ]),
    list([
      'type_indices',
      'type_index',
      'No default values',
      'atom',
      None,
    ]),
    list([
      'types',
      'type',
      "''",
      'atom',
      <class 'object'>,
    ]),
    list([
      'ureybradleys',
      'ureybradleys',
      'No default values',
      'atom',
      None,
    ]),
  ])
# ---
